### PR TITLE
Added a generic middleware routing handler for all request types.

### PR DESCRIPTION
**Pull request by shatyajeet**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
+app.use(function (req, res, next) {
+	if (req.query.url) {
+		var options = {
+			url: req.query.url,
+			method: req.method,
+			headers: req.headers,
+			qs: req.query
+		};
+		if (['PUT', 'POST', 'PATCH'].indexOf(req.method) > -1) {
+			options.body = JSON.stringify(req.body);
```
Comment (mohsen1) [inline index.js]: Why you need json stringify here? I guess we can pass it directly so request

**Comment by mohsen1** [comment]
Overall looks good

File: index.js (hunk 1 above)
Comment (DavidBiesack) [inline index.js]: I'm also curious. I'm not sure of the context in which this is done... will this work for non-JSON requests bodies?

Comment (shatyajeet) [inline index.js]: @mohsen1 I am guessing, setting Content-Type and Accept headers to 'application/json' would help in passing body in json form.

Comment (shatyajeet) [inline index.js]: @DavidBiesack Are you asking about non-JSON requests like multipart?

Comment (mohsen1) [inline index.js]: Yeah I think we should pass the body directly. If it's JSON it will just work. For other content types it might work.

**Comment by shatyajeet** [comment]
Updated the code

**Comment by shatyajeet** [comment]
I have removed the usage of bodyparser as it wasn't necessary. Instead I have used a middleware to pipe the stream into the req.body.

File: index.js (hunk 1 above)
Comment (DavidBiesack) [inline index.js]: yes. I'm curious about non-JSON content. The cors-it proxy can be used for any service, so it should not assume JSON content (could be multi-part form, or images or XML or plain text)...

**Pull request merged**